src/main/scala/util/AsyncResetReg.scala

```diff
@@ -54,26 +55,15 @@ class SimpleRegIO(val w: Int) extends Bundle{
 }
 
 class AsyncResetRegVec(val w: Int, val init: BigInt) extends Module {
-  val io = new SimpleRegIO(w)
+  override def desiredName = s"AsyncResetRegVec_w${w}_i${init}"
 
-  val async_regs = List.tabulate(w) { idx =>
-    val on = if (init.testBit(idx)) 1 else 0
-    Module(new AsyncResetReg(on))
-  }
+  val io = IO(new SimpleRegIO(w))
 
-  val q = for ((reg, idx) <- async_regs.zipWithIndex) yield {
-    reg.io.clk := clock
-    reg.io.rst := reset
-    reg.io.d   := io.d(idx)
-    reg.io.en  := io.en
-    reg.suggestName(s"reg_$idx")
-    reg.io.q
+  val reg = withReset(reset.asAsyncReset)(RegInit(init.U(w.W)))
+  when (io.en) {
+    reg := io.d
   }
-
-  io.q := q.asUInt
-
-  override def desiredName = s"AsyncResetRegVec_w${w}_i${init}"
-
+  io.q := reg
```

Comment:
Omg it’s so much better

Reply:
not that we used it much, but what about the enable signal?